homeassistant/components/config/zwave.py

```diff
@@ -26,13 +25,35 @@ def async_setup(hass):
     hass.http.register_view(ZWaveNodeGroupView)
     hass.http.register_view(ZWaveNodeConfigView)
     hass.http.register_view(ZWaveUserCodeView)
-    hass.http.register_static_path(
-        URL_API_OZW_LOG, hass.config.path(OZW_LOG_FILENAME), False)
+    hass.http.register_view(ZWaveLogView)
     hass.http.register_view(ZWaveConfigWriteView)
 
     return True
 
 
+class ZWaveLogView(HomeAssistantView):
+    """View to read the ZWave log file."""
+
+    url = r"/api/zwave/ozwlog/{lines:\d+}"
+    name = "api:zwave:ozwlog"
+
+    @ha.callback
+    def get(self, request, lines):
+        """Retrieve the lines from ZWave log."""
+        lines = int(lines)
+        hass = request.app['hass']
+        logfilepath = hass.config.path(OZW_LOG_FILENAME)
+
+        with open(logfilepath, 'r') as logfile:
+            data = logfile.read()
```

Comment:
No need to read the whole file into memory if we only want the last N lines.

```python
from collections import deque

with open(logfilepath, 'r') as logfile:
    it = (line.rstrip() for line in logfile)
    if lines == 0:
        result = list(it)
    else:
        result = deque(it, lines)
```

Reply:
Wouldn't it return the *first* N lines?